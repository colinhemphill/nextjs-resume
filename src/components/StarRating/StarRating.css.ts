import { style } from '@vanilla-extract/css';

export const star2 = style({
  opacity: 0.75,
});

export const star3 = style({
  opacity: 0.5,
});
