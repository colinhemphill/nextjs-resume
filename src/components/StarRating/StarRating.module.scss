.secondStar {
  opacity: 0.75;
}
.thirdStar {
  opacity: 0.5;
}