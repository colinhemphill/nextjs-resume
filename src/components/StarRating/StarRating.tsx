import { faStar } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

interface StarRatingProps {
  stars: 1 | 2 | 3;
}

const Star: React.FC<{ className?: string }> = () => {
  return (
    <FontAwesomeIcon
      className="inline h-5 w-5 text-accent-11 dark:text-amber-11"
      icon={faStar}
    />
  );
};

export const StarRating: React.FC<StarRatingProps> = (props) => {
  const { stars } = props;

  return (
    <span className="flex items-center">
      {stars >= 1 && <Star />}
      {stars >= 2 && <Star />}
      {stars >= 3 && <Star />}
    </span>
  );
};
