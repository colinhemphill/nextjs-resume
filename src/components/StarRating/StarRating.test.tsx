import { render } from '@test-utils';
import { describe, expect, test } from 'vitest';
import { StarRating } from './StarRating';

describe('<StarRating />', () => {
  test('Snapshot', async () => {
    const { container } = render(<StarRating stars={3} />);
    expect(container).toMatchSnapshot();
  });
});
