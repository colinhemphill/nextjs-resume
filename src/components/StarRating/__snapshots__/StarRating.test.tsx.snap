// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<StarRating /> > Snapshot 1`] = `
<div>
  <span
    class="flex items-center"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-star inline h-5 w-5 text-accent-11 dark:text-amber-11"
      data-icon="star"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
        fill="currentColor"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-star inline h-5 w-5 text-accent-11 dark:text-amber-11"
      data-icon="star"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
        fill="currentColor"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-star inline h-5 w-5 text-accent-11 dark:text-amber-11"
      data-icon="star"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;
